### PR TITLE
Using scroll-touch-edge to record the distance

### PR DESCRIPTION
**Pull request by darkdh**
cause wheel event will only fire once each scroll since C69

mainly
fix https://github.com/brave/browser-laptop/issues/15114

also
fix https://github.com/brave/browser-laptop/issues/14871
fix https://github.com/brave/browser-laptop/issues/14829
fix https://github.com/brave/browser-laptop/issues/14785

Auditor: @bsclifton

## Submitter Checklist:

- [ ] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [ ] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [ ] Tagged reviewers and labelled the pull request [as needed](https://github.com/brave/browser-laptop/wiki/Pull-request-process).
- [ ] Request a security/privacy review [as needed](https://github.com/brave/handbook/blob/master/development/security.md#how-to-request-a-security-review). (Ask a Brave employee to help if you cannot access this document.)

## Reviewer Checklist:

- [ ] Request a security/privacy review [as needed](https://github.com/brave/handbook/blob/master/development/security.md#how-to-request-a-security-review) if one was not already requested.

Tests

- [ ] Adequate test coverage exists to prevent regressions
- [ ] Tests should be independent and work correctly when run individually or as a suite [ref](https://github.com/brave/browser-laptop/wiki/Code-Guidelines#test-dependencies)
- [ ] New files have MPL2 license header

### CODE REVIEW

**Comment by darkdh** [comment]
don't merge it until master has C69 muon

**Comment by joelwass** [comment]
I believe this logic is broken. If I swipe right on the trackpad with the intention of navigating backwards, 1) the sensitivity is incredibly high, but 2) if i change directions on horizontal scroll and try to "change my mind" and scroll back the other way, your deltaX keeps increasing since I haven't lifted my fingers

**Comment by joelwass** [comment]
I can go ahead and create some changes on top of this PR if you'd like?

**Comment by darkdh** [comment]
no, this is because require chromium 69 which is in 8.1.x. So we have to wait for master upgrading muon version so we can test this PR and merge it

**Comment by darkdh** [comment]
for now, you can try 0.23.x-c69 branch. And create a PR on top of it if you want

File: app/renderer/components/main/main.js [hunk 1]
```diff
@@ -275,16 +257,23 @@ class Main extends React.Component {
...
+        const distanceThresholdX = getSetting(settings.SWIPE_NAV_DISTANCE)
+        const percent = Math.abs(deltaX) / distanceThresholdX
+        time = (new Date()).getTime() - startTime
+        if (deltaX > 0) {
+          if (percent > 1) {
+            appActions.swipedRight(1)
+          } else {
+            appActions.swipedRight(percent)
+          }
+          deltaX += 1
```
Comment (joelwass) [inline app/renderer/components/main/main.js]: what's the intended purpose of this line and line 276? that seems to cause the issue i mentioned in the discussion

Comment (darkdh) [reply app/renderer/components/main/main.js]: again, that is intentional for `features::kTouchpadAndWheelScrollLatching` enabled in c69 and master does't have the compatible version of muon now.
could you try 0.23.x-c69 branch first?

Comment (darkdh) [reply app/renderer/components/main/main.js]: did you delete `node_modules/electron-prebuilt` and redo a npm install?
If you didn't, that means you don't have the correct muon version.

Comment (darkdh) [reply app/renderer/components/main/main.js]: you can also try https://github.com/brave/browser-laptop/releases/tag/v0.23.202beta which is just built from that branch and contains this fix

Comment (joelwass) [reply app/renderer/components/main/main.js]: yes i rm rf'd node modules on the `0.23.x-c69` branch and the issue still occurs. Also the issue i mentioned still occurs in https://github.com/brave/browser-laptop/releases/tag/v0.23.202beta

would you like me to try to fix?

Comment (darkdh) [reply app/renderer/components/main/main.js]: I believe the problem you talked about is no way to cancel the navigation once you reach 100%. you can try to fix that. But I don't see so called incredibly high sensitivity

Comment (darkdh) [reply app/renderer/components/main/main.js]: I'm going to emit more attributes from muon for this event so I can update the precise units

Comment (darkdh) [reply app/renderer/components/main/main.js]: you can try 0.23.x-c69 again now. new muon build is ready

**Comment by bsclifton** [comment]
Closing as `wontfix` - this should be fixed in the new version of Brave:
https://brave.com/download/

**Pull request closed without merging**